quipucords/scanner/network/inspect_callback.py

```diff
@@ -136,11 +162,12 @@ def event_callback(self, event_dict=None):
         okay = ["runner_on_ok", "runner_item_on_ok"]
         failed = ["runner_on_failed", "runner_item_on_failed"]
         unreachable = ["runner_on_unreachable"]
-        runner_ignore = [
-            "runner_on_skipped",
-            "runner_item_on_skipped",
-            "runner_on_start",
-        ]
+        skipped = ["runner_on_skipped", "runner_item_on_skipped"]
+        runner_ignore = ["runner_on_start"]
+        if not self._collect_skipped_tasks_enabled:
+            runner_ignore.extend(skipped)
+            skipped = []
```

Comment:
I find this entire approach puzzling. That seems like an attempt at performance improvement, to catch events we are not interested in early and exit the function.

But if we are not interested in event, then we are going to compare it to three short lists (four with this PR), before it falls through `elif`s and function exits. How much time could be saved by skipping three `in` checks, even if there are thousands of events? And is this really Discovery hotspot?

To add insult to injury, there are Ansible events that are not in `runner_ignore`, so we pay the price of three `in` checks for them anyway. These events are at least `playbook_on_play_start`, `playbook_on_stats`, `playbook_on_task_start` and `verbose`. In my test run, these constituted about 52% of all Ansible events.

Reply:
I mindlessly backported these checks from the old code.